hydromt/data_adapter/harmonization_settings.py

```diff
@@ -0,0 +1,19 @@
+"""Settings used to harmonize data."""
+from typing import Any, Dict, Union
+
+from pydantic import BaseModel, ConfigDict, Field
+
+
+class HarmonizationSettings(BaseModel):
```

Comment:
The problem is that you also need these in the `MetaDataResolver`. (although there we also really need the actual settings for harmonization). I can add these (`extent`, `attrs` and `meta`) to the `DataAdapter` model? Or to avoid copying these over, we can also work with backrefs like in the `ModelComponent` class, then we just save these settings in `DataSource`.

Reply:
So perhaps:
```yaml
cooldata:
  ...
  data_adapter:
    unit_add:
      variable1: 2.0
      variable3: -6.5
  meta:
    myproperty: 'important'
    myotherproperty: 'less_so'
  attrs:
    my_attr_1: 3
    ...
  ...
```